lnwallet/channel.go

```diff
@@ -1957,6 +1962,8 @@ func newBreachRetribution(chanState *channeldb.OpenChannel, stateNum uint64,
 //
 // NOTE: This MUST be run as a goroutine.
 func (lc *LightningChannel) closeObserver(channelCloseNtfn *chainntnfs.SpendEvent) {
+	defer lc.wg.Done()
```

Comment:
We don't necessarily always need to wait on this when stopping the channel. With recent changes, this goroutine may outlive the lifetime of the channel itself.